### PR TITLE
Fixes related to channels

### PR DESCRIPTION
**Pull request by noxiouz**
I will fill the discription later.

### CODE REVIEW

File: core/context.go [hunk 1]
```diff
@@ -102,7 +104,7 @@ func (ctx *Context) Close() {
 	log.Info("shutting down IPVS context")
 
 	// This will also shutdown the pulse notification sink goroutine.
-	close(ctx.pulseCh)
+	close(ctx.stopCh)
```
Comment (kobolog) [inline core/context.go]: I don't get it. It was working fine already: `close(ctx.pulseCh)` will break the for loop in `notificationSink()`, why do we need another channel here?

File: pulse/pulse.go [hunk 2]
```diff
@@ -83,9 +83,12 @@ func (p *Pulse) Loop(id ID, pulseCh chan Update) {
 	for {
 		select {
 		case <-time.After(interval):
+			select {
 			// Recalculate metrics and statistics and send them to Context.
-			pulseCh <- Update{id, p.metrics.Update(p.driver.Check())}
-
+			case pulseCh <- Update{id, p.metrics.Update(p.driver.Check())}:
```
Comment (kobolog) [inline pulse/pulse.go]: Here, too – what's the reason for another channel? Can't we do:

```go
select {
case pulseCh <- Update{...}
default:
}
```

File: core/context.go (hunk 1 above)
Comment (noxiouz) [inline core/context.go]: You close pulseCh, but Loop goroutine will try to send an update to it. It's panic. As there's no way to find out if the channel is closed, you cannot close it or make sure that no one will use the channel for writting after close.

File: pulse/pulse.go (hunk 2 above)
Comment (noxiouz) [inline pulse/pulse.go]: You send will fail if the consumer is handling your previous data at that time

File: core/context.go (hunk 1 above)
Comment (kobolog) [inline core/context.go]: Omg, Go sucks even more than I thought.

**Comment by kobolog** [comment]
I'm gonna read up a little bit more about how Go channels work before merging this. Not because this PR is bad or something, but because I wanna understand what I'm merging better. Also, could you please rebase it?

**Comment by noxiouz** [comment]
rebased

**Comment by kobolog** [comment]
Hey, can you please rebase it so I can merge? Totally forgot about it!

**Comment by noxiouz** [comment]
done

**Pull request merged**